### ISSUE TITLE
[Web] Error "Can't access output tensor data. error code = 1."

### ISSUE THREAD

**Issue by xiaohk** [microsoft/onnxruntime#14493]
### Describe the issue

I use `onnxmltools` to convert a LightGBM model to an onnx file through `convert_lightgbm`. I can run this model using the ONNX runtime in Python. However, I got an error  "Can't access output tensor data. error code = 1." when I tried to inference it using ONNX runtime web.

### To reproduce

The model can be found in this [dropbox](https://www.dropbox.com/s/voiow5v071pq7hy/model.onnx?dl=0).

I followed [the example](https://github.com/microsoft/onnxruntime-inference-examples/blob/main/js/quick-start_onnxruntime-web-bundler/main.js) to run ONNX runtime in browser.

```typescript
try {
  // Create a new session and load the LightGBM model
  const session = await ort.InferenceSession.create(modelUrl);

  const x = Float32Array.from(new Array(31).fill(0));

  // Prepare feeds. use model input names as keys.
  const xTensor = new ort.Tensor('float32', x, [1, 31]);
  const feeds = { float_input: xTensor };

  // Feed inputs and run
  const results = await session.run(feeds);

  this.message = `Success: ${results}`;
} catch (e) {
  this.message = `Failed: ${e}.`;
}
```

My ONNX runtime set up:

```typescript
import wasm from 'onnxruntime-web/dist/ort-wasm.wasm?url';
import wasmThreaded from 'onnxruntime-web/dist/ort-wasm-threaded.wasm?url';
import wasmSimd from 'onnxruntime-web/dist/ort-wasm-simd.wasm?url';
import wasmSimdThreaded from 'onnxruntime-web/dist/ort-wasm-simd-threaded.wasm?url';

import modelUrl from './model.onnx?url';

ort.env.wasm.wasmPaths = {
  'ort-wasm.wasm': wasm,
  'ort-wasm-threaded.wasm': wasmThreaded,
  'ort-wasm-simd.wasm': wasmSimd,
  'ort-wasm-simd-threaded.wasm': wasmSimdThreaded
};

```

### Urgency

Quite urgent, have a paper deadline this Friday

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.13.1

### Execution Provider

WASM

**Comment by RyanUnderhill** [comment]
@fs-eire Can you help or assign this one?

**Comment by shalvamist** [comment]
Hi Xiaohk,

Thanks for the detailed report of the issue.
I think since there are 2 outputs you need to use the 'fetches' option in run API. 
I was able to run the model using this modification - 
```
      try {
          const session = await ort.InferenceSession.create('./model.onnx');

          // prepare inputs. a tensor need its corresponding TypedArray as data
          const dataA = Float32Array.from(new Array(31).fill(0));
          const tensorA = new ort.Tensor('float32', dataA, [1, 31]);

          let dataB = new BigInt64Array([ BigInt(Number.MAX_SAFE_INTEGER) ]);
          let tensorB = new ort.Tensor('int64', dataB, [1]);

          let map1 = new Map();

          // prepare feeds. use model input names as keys.
          const feeds = { float_input: tensorA};

          // prepare feeds. use model input names as keys.
          const fetches = { label: tensorB, probabilities: map1};

          // feed inputs and run
          const results = await session.run(feeds, fetches);

          // read from results
          document.write(`Results: ${results.label} ${results.probabilities}`);

      } catch (e) {
          document.write(`failed to inference ONNX model: ${e}.`);
      }
  }
```
Please try this approach and let me know if it resolves the problem.

**Comment by xiaohk** [comment]
Thank you so much for your quick rely @shalvamist and @RyanUnderhill.

I tried your code and my error is gone 👍. However, the `results` object only has a `label` attribute, and it doesn't has the `probabilities` attribute. Do you have any suggestions to debug it?

Here is my code:

```ts
    try {
      // Create a new session and load the LightGBM model
      const session = await ort.InferenceSession.create(modelUrl);

      const x = Float32Array.from(new Array(31).fill(0));

      // Prepare feeds, use model input names as keys.
      const xTensor = new ort.Tensor('float32', x, [1, 31]);
      const feeds = { float_input: xTensor };

      let dataB = new BigInt64Array([BigInt(Number.MAX_SAFE_INTEGER)]);
      let tensorB = new ort.Tensor('int64', dataB, [1]);
      let map1 = new Map();

      // prepare feeds. use model input names as keys.
      const fetches = { label: tensorB, probabilities: map1 };

      // Feed inputs and run
      const results = await session.run(feeds, fetches);

      // Read from results
      // const probs = results.probabilities.data;
      console.log(results);

      this.message = `Success: ${results}`;
    } catch (e) {
      this.message = `Failed: ${e}.`;
    }
```

Here is the printed out `results`

![image](https://user-images.githubusercontent.com/15007159/215851589-8cb5161c-8e14-42ca-bd97-70fb7e90ecb0.png)

**Comment by fs-eire** [comment]
The model uses non-tensor type (probabilities is type of `sequence<map<int64,float32>>`) as its input/output and onnxruntime-web does not support those types yet. a quick fix is to re-export the model with options that disable the non-tensor types.

**Comment by xiaohk** [comment]
Thank you all for helping out! I eventually solved this issue by using XGBoost instead of LightGBM. I believe @fs-eire's suggestion will work.

On a related note, we encountered this issue when we were developing [WebSHAP](https://poloclub.github.io/webshap/), which is now public! WebSHAP is a JavaScript library that enables users to explain any ML models in their browser. We created three application examples using WebSHAP (explaining XGBoost, CNNs, and transformer models). We used ONNX Web Runtime for two examples. I was impressed to see how easy it was to use ONNX Web Runtime and everything just worked! Thank you all for creating and maintaining ONNX Web Runtime!! ❤️ 


![webshap-demo](https://user-images.githubusercontent.com/15007159/228648090-67fa3efa-a091-43b2-8940-cbadd3730e12.gif)
